File: Algorithms/src/andy/level1/ClawCrane.java
package andy.level1;
import java.util.Stack;

public class ClawCrane {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        Stack<Integer> stack = new Stack<>();
        
        for(int item:moves) {
            item--;
            for(int i=0;i<board.length;i++) {
                if(board[i][item] != 0) {
                    if(!stack.empty() && stack.peek() == board[i][item]) {
                        stack.pop();
                        answer += 2;
                    }
                    else 
                        stack.push(board[i][item]);
                    board[i][item] = 0;
                    break;                    
                }
            }
            
        }
        
        return answer;
    }
}

File: Andy/src/level1/javascript/ClawCrane.js
// 크레인 인형뽑기 게임
function solution(board, moves) {
    var answer = 0;
    var arr = Array();
    board.pickup = function(idx) {
        for(let i=0;i<board.length;i++) {
            if(this[i][idx]>0) {
                let tmp = this[i][idx];
                this[i][idx] = 0;
                return tmp;
            }
        }
        return 0;
    }

	moves.forEach(function(item) {
		item--;
        let data = board.pickup(item);
        if(data)  {
            if(arr.length && arr[arr.length-1] == data) {
                arr.pop();
                answer += 2;
            }
            else 
                arr.push(data);
        }
	});
    return answer;    
  
}